crates/runtime/src/sys/unix/signals.rs

```diff
@@ -15,10 +31,11 @@ static mut PREV_SIGBUS: MaybeUninit<libc::sigaction> = MaybeUninit::uninit();
 static mut PREV_SIGILL: MaybeUninit<libc::sigaction> = MaybeUninit::uninit();
 static mut PREV_SIGFPE: MaybeUninit<libc::sigaction> = MaybeUninit::uninit();
 
-pub unsafe fn platform_init() {
-    if cfg!(miri) {
-        return;
-    }
+pub unsafe fn platform_init(macos_use_mach_ports: bool) {
+    // Either mach ports shouldn't be in use or we shouldn't be on macOS,
+    // otherwise the `machports.rs` module should be used instead.
+    assert!(!macos_use_mach_ports || !cfg!(target_os = "macos"));
```

Comment:
Why not make this a compile-time check instead, then?

Reply:
Could you elaborate? Do you mean asserting that this file isn't used on macos? (it's used on macos intentionally so wanted to clarify)

Reply:
Oh, sorry... I was just looking at the `cfg!` part but clearly `machos_use_mach_ports` is runtime-defined. Never mind!